### PR TITLE
Remove AnimatedSimulation

### PR DESCRIPTION
**Pull request by abarth**
This patch folds the functionality from AnimatedSimulation into Timeline.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie One less animation class!

**Comment by Hixie** [comment]
Will review later. This one requires more care than my usual weekend reviews. ;-)

**Comment by abarth** [comment]
Yeah, no worries.  It's actually pretty mechanical.  The diff just looks ugly.  Basically, I moved these classes into one file and inlined the functions.

**Comment by abarth** [comment]
Fixed #1167

File: sky/packages/sky/lib/src/animation/timeline.dart [hunk 1]
```diff
@@ -5,17 +5,89 @@
...
+      return;
+
+    _startTime = null;
+
+    if (_animationId != null) {
+      scheduler.cancelAnimationFrame(_animationId);
+      _animationId = null;
+    }
+
+    // We take the _completer into a local variable so that !isTicking
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/timeline.dart]: this comment's grammar is suspect.

File: sky/packages/sky/lib/src/animation/timeline.dart [hunk 2]
```diff
@@ -34,24 +106,41 @@ class _TweenSimulation extends Simulation {
   bool isDone(double timeInSeconds) => timeInSeconds > _durationInSeconds;
 }
 
+typedef TimelineCallback(double value);
+
 /// A timeline for an animation
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/timeline.dart]: This comment doesn't tell me what this class does. Specifically, both "timeline" and "animation" are far too vague.

File: sky/packages/sky/lib/src/animation/timeline.dart [hunk 3]
```diff
@@ -5,17 +5,89 @@
...
+  }
+
+  void _scheduleTick() {
+    assert(isTicking);
+    assert(_animationId == null);
+    _animationId = scheduler.requestAnimationFrame(_tick);
+  }
+}
+
+/// A simulation that varies from [begin] to [end] over [duration] using [curve]
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/timeline.dart]: should we rename "simulation"? It's not clear to me that something as simple as "apply a function to a variable" should be called a "simulation".

How does this class differ from an AnimatedValue<double>? As far as I can tell, all the logic in this class can be summed up in two lines:
```
_tween = new AnimatedValue<double>(begin, end: end, curve: curve);
// ...
_tween.setProgress(t, Direction.forward);
```

File: sky/packages/sky/lib/src/animation/timeline.dart [hunk 4]
```diff
@@ -5,17 +5,89 @@
...
-import 'package:sky/src/animation/curves.dart';
 import 'package:sky/src/animation/animated_value.dart';
-import 'package:sky/src/animation/animated_simulation.dart';
+import 'package:sky/src/animation/curves.dart';
+import 'package:sky/src/animation/scheduler.dart';
+
+typedef _TickerCallback(Duration elapsed);
+
+/// Calls its callback once per animation frame
+class _Ticker {
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/timeline.dart]: This class (_Ticker) actually seems like a legitimate primitive. Maybe it should be in the scheduler.dart file or in its own ticker. dart file. It feels like a lower level than the rest of the animation system.

File: sky/packages/sky/lib/src/animation/timeline.dart (hunk 3 above)
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/timeline.dart]: maybe say here what you said to me in person about it being an adapter

File: sky/packages/sky/lib/src/animation/timeline.dart [hunk 5]
```diff
@@ -5,17 +5,89 @@
...
+  void _scheduleTick() {
+    assert(isTicking);
+    assert(_animationId == null);
+    _animationId = scheduler.requestAnimationFrame(_tick);
+  }
+}
+
+/// A simulation that varies from [begin] to [end] over [duration] using [curve]
 class _TweenSimulation extends Simulation {
   final double _durationInSeconds;
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/timeline.dart]: fields after the constructor

File: sky/packages/sky/lib/src/animation/timeline.dart (hunk 2 above)
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/timeline.dart]: maybe rename this class. How is it a timeline?

File: sky/packages/sky/lib/src/animation/timeline.dart [hunk 6]
```diff
@@ -34,24 +106,41 @@ class _TweenSimulation extends Simulation {
...
-    _animation.value = newValue;
+    _value = newValue;
+    _onTick(_value);
   }
 
   /// Whether the timeline is currently animating
-  bool get isAnimating => _animation.isAnimating;
+  bool get isAnimating => _ticker.isTicking;
+
+  /// Start ticking the given simulation once per frame
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/timeline.dart]: move _start and _stop below animateTo() and fling().

File: sky/packages/sky/lib/src/animation/timeline.dart [hunk 7]
```diff
@@ -59,18 +148,27 @@ class Timeline {
...
 
   /// Stop animating the timeline
   void stop() {
-    _animation.stop();
+    _simulation = null;
+    _ticker.stop();
   }
 
   /// Gives the given simulation control over the timeline
   Future fling(Simulation simulation) {
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/timeline.dart]: Is fling() a good term?

**Comment by Hixie** [comment]
I think Timeline should be Animator.

LGTM

**Comment by Hixie** [comment]
I would also accept AnimationDriver or SimulationDriver or something like that.

**Pull request merged**